tools-local/Microsoft.DotNet.Build.Tasks.Local/UploadClient.cs

```diff
@@ -142,6 +141,14 @@ public string EncodeBlockIds(int numberOfBlocks, int lengthOfId)
                     var req = new HttpRequestMessage(HttpMethod.Put, blockListUploadUrl);
                     req.Headers.Add(AzureHelper.DateHeaderString, dt1.ToString("R", CultureInfo.InvariantCulture));
                     req.Headers.Add(AzureHelper.VersionHeaderString, AzureHelper.StorageApiVersion);
+                    if(!string.IsNullOrEmpty(DetermineContentTypeBasedOnFileExtension(filePath)))
```

Comment:
should probably put `DetermineContentTypeBasedOnFileExtension` in a variable and use it below, so we don't need two calls.